UMA-Bots/Large-Relays/src/agent.spec.ts

```diff
@@ -0,0 +1,191 @@
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(RANDOM_EVENT_ABI, TEST_SPOKEPOOL_ADDR, [RANDOM_ADDRESSES[0], "120"]);
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a small (less than threshold) amount of non-specified tokens are bridged via whitelisted SpokePool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(FILLED_RELAY_EVENT, TEST_SPOKEPOOL_ADDR, passParams("21", RANDOM_ADDRESSES[2]));
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a small (less than threshold) amount of specified tokens are bridged via whitelisted SpokePool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(FILLED_RELAY_EVENT, TEST_SPOKEPOOL_ADDR, passParams("21", MOCK_ERC20_ADDR));
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a large amount of non-specified tokens are bridged via whitelisted SpokePool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(
+        FILLED_RELAY_EVENT,
+        TEST_SPOKEPOOL_ADDR,
+        passParams(MOCK_THRESHOLD_ERC20.toString(), RANDOM_ADDRESSES[0])
+      );
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns a finding if a large amount of specified tokens are bridged via whitelisted SpokePool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(
+        FILLED_RELAY_EVENT,
+        TEST_SPOKEPOOL_ADDR,
+        passParams(MOCK_THRESHOLD_ERC20.toString(), MOCK_ERC20_ADDR)
+      );
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([expectedFinding(MOCK_THRESHOLD_ERC20.toString(), MOCK_ERC20_ADDR)]);
+  });
+
+  it("returns N findings when N large relays occurs from the whitelisted SpokePool address", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(
+        FILLED_RELAY_EVENT,
+        TEST_SPOKEPOOL_ADDR,
+        passParams(MOCK_THRESHOLD_ERC20.toString(), MOCK_ERC20_ADDR)
+      )
+      .addEventLog(
+        FILLED_RELAY_EVENT,
+        TEST_SPOKEPOOL_ADDR,
+        passParams(MOCK_THRESHOLD_ERC20.toString(), MOCK_ERC20_ADDR_2)
+      )
+      .addEventLog(RANDOM_EVENT_ABI, TEST_SPOKEPOOL_ADDR, [RANDOM_ADDRESSES[0], "120"]) // Non-whitelisted SpokePool events shall be ignored
```

Comment:
This "non-whitelisted" basically means "other", correct?

Reply:
Yep

Reply:
can be unofficial? according to the convo above

Reply:
I just rephrased it. Please check if it is more readable now 😃